ring/ring.go

```diff
@@ -285,21 +285,7 @@ func (r *Ring) starting(ctx context.Context) error {
 
 func (r *Ring) loop(ctx context.Context) error {
 	// Update the ring metrics at start of the main loop.
-	r.updateRingMetrics()
-	go func() {
-		// Start metrics update ticker to update the ring metrics.
-		ticker := time.NewTicker(10 * time.Second)
-		defer ticker.Stop()
-
-		for {
-			select {
-			case <-ticker.C:
-				r.updateRingMetrics()
-			case <-ctx.Done():
-				return
-			}
-		}
-	}()
+	r.updateRingMetrics(Different)
```

Comment:
It should run it taking the read lock.

Reply:
Re-thinking it, may just be safer to take the read lock directly in `updateRingMetrics()`. Look at where `updateRingMetrics()` is called, it looks an easier change.

Reply:
Added locking around the initial metrics - since it's called just once and also due to upcoming PR 109, I used write lock. We discussed the locking IRL and opted to keep the locking outside updateRingMetrics